test/messenger.test.ts

```diff
@@ -0,0 +1,177 @@
+import Messenger from '../src/messenger';
+import { EmitterMessage, WorkerMessage } from '../src/types';
+
+describe('Messenger', () => {
+  describe('constructor', () => {
+    it('creates and connects to a shared worker', () => {
+      const sharedWorkerUrl = 'http://foo.bar.org';
+      const mockStart = jest.fn();
+      SharedWorker = jest.fn(
+        () =>
+          ({
+            port: {
+              start: mockStart,
+              onmessage: undefined,
+            },
+          } as any)
+      );
+      const handleWorkerSpy = jest.spyOn(
+        Messenger.prototype,
+        'handleWorkerMessage'
+      );
+      handleWorkerSpy.mockImplementation(jest.fn());
+      const messenger = new Messenger({
+        sharedWorkerUrl,
+      });
+      expect(SharedWorker).toHaveBeenCalledWith(sharedWorkerUrl);
+      expect(mockStart).toHaveBeenCalled();
+      (messenger['worker'] as any)['port']['onmessage']({ data: 'foo' });
```

Comment:
Q: How come this requires a cast to any? Can we just do the following here?

```suggestion
      messenger.worker.port.onmessage({ data: 'foo' });
```

Reply:
same re: typescript being needy 😕